### PR TITLE
Added Generic API for PIN

### PR DESCRIPTION
**Pull request by radhay-samagra**
## Description

Added Generic APIs for PIN

### Changes

- Added Login By Pin
- Added Signup By Pin
- Added CHange Password By Pin
- Added refresh token API
- Fixed issues with generic env variables causing docker-compose issues for application IDs

## How to test

Describe the steps required to test the changes proposed in the pull request.

Please consider using the closing keyword if the pull request is proposed to
fix an issue already created in the repository
(https://help.github.com/articles/closing-issues-using-keywords/)

### CODE REVIEW

File: src/api/api.controller.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ import { ConfigResolverService } from './config.resolver.service';
 import { FusionauthService } from './fusionauth/fusionauth.service';
 import { OtpService } from './otp/otp.service';
 import { SMSResponse } from './sms/sms.interface';
+import { RefreshRequest } from '@fusionauth/typescript-client/build/src/FusionAuthClient';
```
Comment (radhay-samagra) [inline src/api/api.controller.ts:24]: You should not expose fusionauth interfaces here. This is a leaky abstraction. Let's try to have the boundaries defined for this service and the FA service.

Comment (radhay-samagra) [reply src/api/api.controller.ts:24]: https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/

Comment (choxx) [reply src/api/api.controller.ts:24]: Have made improvements in this [commit](https://github.com/Samagra-Development/user-service/pull/31/commits/7c4105763cbe1736e8b9701ad4b1f2f484afd7af). Please check.

File: src/api/api.controller.ts [hunk 2]
```diff
@@ -60,12 +65,31 @@ export class ApiController {
...
       } 
       const status: SignupResponse = await this.apiService.login(user, authHeader);
       return status;
   }
+
+  @Post('login/pin')
+  async loginByPin(@Body() user: any, @Headers('authorization') authHeader): Promise<any> {
+      const encStatus = this.configResolverService.getEncryptionStatus(user.applicationId);
+      const encodedBase64Key = this.configResolverService.getEncryptionKey(user.applicationId);
+      const parsedBase64Key = encodedBase64Key === undefined? CryptoJS.enc.Base64.parse('bla'): CryptoJS.enc.Base64.parse(encodedBase64Key);
```
Comment (ChakshuGautam) [inline src/api/api.controller.ts:81]: Keep the `bla` as a salt in the env for the application config.

File: src/api/api.controller.ts [hunk 3]
```diff
@@ -78,20 +102,39 @@ export class ApiController {
...
   }
 
+  @Post('changePin')
+  async updatePin(
+    @Body() data: { loginId: string, password: string },
+    @Headers('authorization') authHeader,
+    @Headers('x-application-id') applicationId
+  ): Promise<SignupResponse> {
+    const encodedBase64Key = this.configResolverService.getEncryptionKey(applicationId);
+    const parsedBase64Key = encodedBase64Key === undefined? CryptoJS.enc.Base64.parse('bla'): CryptoJS.enc.Base64.parse(encodedBase64Key);
```
Comment (ChakshuGautam) [inline src/api/api.controller.ts:120]: Here as well.

**Pull request merged**